datafusion/physical-expr/src/math_expressions.rs

```diff
@@ -497,6 +497,91 @@ pub fn log(args: &[ArrayRef]) -> Result<ArrayRef> {
     }
 }
 
+/// Truncate(numeric, decimalPrecision) and trunc(numeric) SQL function
+pub fn trunc(args: &[ArrayRef]) -> Result<ArrayRef> {
+    if args.len() != 1 && args.len() != 2 {
+        return Err(DataFusionError::Internal(format!(
+            "truncate function requires one or two arguments, got {}",
+            args.len()
+        )));
+    }
+
+    //if only one arg then invoke toolchain trunc(num) and precision = 0 by default
+    //or then invoke the compute_truncate method to process precision
+    let mut _precision = ColumnarValue::Scalar(Int32(Some(0)));
+
+    let num = &args[0];
+    if args.len() == 2 {
+        _precision = ColumnarValue::Array(args[1].clone());
+    }
```

Comment:
The use of `_` as a prefix of a variable is used to tell the compiler it can be unused. In this case I think it is used and thus the variable should be named `precision`

A more "idiomatic" rust way (and avoiding the `mut`) of writing this might be
```suggestion
    let precision = if args.len() == 1 {
      ColumnarValue::Scalar(Int32(Some(0)))
     } else {
       ColumnarValue::Array(args[1].clone())
    };
```